src/main.js

```diff
@@ -551,7 +569,13 @@ async function resolveNlsConfiguration() {
 		// Try to use the app locale. Please note that the app locale is only
 		// valid after we have received the app ready event. This is why the
 		// code is here.
-		let appLocale = app.getLocale();
+
+		// The ternary and ts-ignore can both be removed once Electron
+		// officially adopts the getSystemLocale API.
+		// Ref https://github.com/electron/electron/pull/35697
+		let appLocale = 'getSystemLocale' in app ?
+			// @ts-ignore API not yet available in the official Electron
+			app.getSystemLocale() : app.getLocale();
```

Comment:
Why not remove the country code from `app.getSystemLocale()` before passing to `localization.contribution.ts` ?

Reply:
It is needed in some cases. Some language packs have country codes, and others don't.

Reply:
Maybe the call could be surrounded with `perf.mark('code/willGetSystemLocale');` and `perf.mark('code/didGetSystemLocale');` to ensure we track the time this needs.

Reply:
Both the API are just getters, they don't calculate the values on demand rather fetch the values that were calculated by Electron on startup before user script starts executing, so these marks wouldn't add any value to the actual time spent on those. We could instead have a mark on the native side if we are interested in the actual cost.

Reply:
I have removed the language code in some cases to improve the language pack recommender. See below for the if statement